source/MaterialXRuntime/Private/PvtStage.cpp

```diff
@@ -105,15 +105,20 @@ void PvtStage::disposePrim(const PvtPath& path)
     }
 
     // Make sure the prim has no connections.
-    for (RtAttribute attr : prim->getAttributes())
+    for (size_t i=0; i<prim->numInputs(); ++i)
```

Comment:
I'm probably wrong as you've done this a bunch, but should this be using getInputs() and getOutputs() instead?

Reply:
Again it's mainly a matter of taste. Here `prim` is a PvtPrim (private API). I could have used getInputs() to iterate on the internal vector of PvtObject pointers, as you suggest, but then I get a PvtObject* so need to cast that to a PvtInput* myself. This cast is done by the getInput(index) method. So the number of lines of code is the same.